src/App.jsx

```diff
@@ -1,34 +1,52 @@
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
 import React from 'react';
-import './index.css';
-import { TopNavBar } from './components/TopNavBar/TopNavBar.jsx';
+import './index.scss';
+import { getFlights } from './services/flightsDataService'
+import { TopNavBar } from './components/TopNavBar/TopNavBar';
 import { SideNavBar } from './components/SideNavBar/SideNavBar';
 import { FlightsTable } from './components/FlightsTable/FlightsTable';
 
 export const App = () => {
-  const [flights, setFlights] = useState([
-    { id: 1, origin: 'DEN', destination: 'BOS'},
-    { id: 2, origin: 'DEN', destination: 'FLL'},
-    { id: 3, origin: 'DEN', destination: 'PVD'},
-    { id: 4, origin: 'ORD', destination: 'PVD'},
-    { id: 5, origin: 'PBI', destination: 'BOS'},
-  ]);
+  const [flights, setFlights] = useState(null);
+  const [origins, setOrigins] = useState([]);
+  const [originFilter, setOriginFilter] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isError, setIsError] = useState(false);
 
-  const origins = [...new Set(flights.map((flight) => flight.origin))];
+  useEffect(() => {
+    // Adding a timeout just to show off the loading effect 
+    setTimeout(() => {
+      const data = getFlights()
+        .then( data => {
+        setFlights(data);
+        setOrigins([...new Set(data.map((dataPoint) => dataPoint.origin))]);
+        setIsLoading(false);
+      })
+      .catch(err => {
+        setIsError(true);
+        setIsLoading(false);
+      })
+      }, 1000)
+    }, []
```

Comment:
Your `useEffect` needs to clean up after itself.

If the component gets unmounted, it will still make the API request and try to do something with it, causing a memory leak.

Utilize both the techniques here:
https://blog.logrocket.com/understanding-react-useeffect-cleanup-function/

Reply:
Done. FYI - I took this down a rabbit hole with a slightly different implementation on the signal, and discovered that the cancel token approach is now depreciated :) https://axios-http.com/docs/cancellation